cronenbroguelike/game.py

```diff
@@ -47,64 +47,68 @@ def execute(self):
-            name="player",
-        )
-        _G.player.log_stats = True
-
-        # Resets just_died flag.
-        _G.add_event(_ResetDiedFlag(), "pre")
+def _restart(config):
+    # Resets all global state (clears event queues, etc.).
+    _G.reset()
+
+    # Creates the player character and ensures game will restart upon death.
+    _G.player = actor.create_actor(
+        health=10,
+        psyche=10,
+        strength=10,
+        stamina=10,
+        name="player",
+    )
+    _G.player.log_stats = True
 
-        # Creates a small dungeon.
-        level = floor.Floor.generate("cathedral", config["num_rooms"])
+    # Resets just_died flag.
+    _G.add_event(_ResetDiedFlag(), "pre")
 
-        # Places a monster in a random room.
-        level.random_room().add_character(npcs.fish_man())
+    # Creates a small dungeon.
+    level = floor.Floor.generate("cathedral", config["num_rooms"])
 
-        # Places an NPC in a random room.
-        level.random_room().add_character(npcs.mad_librarian())
+    # Places a monster in a random room.
+    level.random_room().add_character(npcs.fish_man())
 
-        # Places a cool NPC in a random room.
-        level.random_room().add_character(npcs.smokes_man())
+    # Places an NPC in a random room.
+    level.random_room().add_character(npcs.mad_librarian())
 
-        # Places the player.
-        level.random_room().add_character(_G.player)
+    # Places a cool NPC in a random room.
+    level.random_room().add_character(npcs.smokes_man())
 
-        # Starts it up.
-        _get_random_start()
-        adventurelib.set_context("start_game")
-        adventurelib.set_context(None)
-        with _poll_events(poll_before=True, poll_after=True):
-            commands.enter_room(_G.player.current_room)
+    # Places the player.
+    level.random_room().add_character(_G.player)
 
-    _restart(None)
+    # Starts it up.
+    _get_random_start()
+    adventurelib.set_context("start_game")
+    adventurelib.set_context(None)
+    with _poll_events(poll_before=True, poll_after=True):
+        commands.enter_room(_G.player.current_room)
 
 
-def main():
-    game_config = _load_config()
-    if game_config.get("extra_commands") or game_config.get("random_run"):
+def run_game(config):
+    num_random_actions = 0
+    if config.get("extra_commands") or config.get("random_run"):
         from . import extra_commands
     while True:
         try:
-            _start_game(game_config)
+            _restart(config)
             adventurelib.say("")  # Necessary for space before first prompt.
-            if game_config.get("random_run"):
-                while True:
+            if config.get("random_run"):
+                while num_random_actions < config.get("random_run"):
                     adventurelib._handle_command("random")
+                    num_random_actions += 1
+                if num_random_actions >= config.get("random_run"):
+                    break
             adventurelib.start()
```

Comment:
Should `adventurelib.start()` be under an `else` clause? It looks like the game will start to solicit player input at this point, no?

Reply:
The `break` above breaks out of the outer-most while loop, so we skip past that call. But maybe worth changing the code to be more clear?

Reply:
No, I just wasn't reading right; I think the enclosing `try` messed with me. This is good, then!